core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java

```diff
@@ -76,7 +75,12 @@ public ExistingElementMutationImpl(T element) {
 
     @Override
     public ElementMutation<T> deleteProperty(Property property) {
-        FetchHint.checkFetchHints(getElement().getFetchHints(), EnumSet.of(FetchHint.PROPERTIES, FetchHint.PROPERTY_METADATA));
+        if (!element.getFetchHints().isIncludeAllProperties()) {
```

Comment:
Same comment as above. Do we need to have loaded all of the properties to delete one? What if we loaded the one you are trying to delete?

Reply:
done